pkg/pod/status.go

```diff
@@ -116,6 +118,12 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1alpha1.TaskRun, pod *core
 	trs.Steps = []v1alpha1.StepState{}
 	trs.Sidecars = []v1alpha1.SidecarState{}
 
+	ts := v1beta1.TaskSpec{}
```

Comment:
We might want to store this into the TaskRun status when we first fetch it, i.e. https://github.com/tektoncd/pipeline/blob/9344e4f2291bb5b9a35f50f7549c520e02c8bf56/pkg/reconciler/taskrun/taskrun.go#L207-L208

We could then modify that could to fetch the task only if not available in the status, and else use that from status

Reply:
Whoa, yeah thanks. Do you think it's enough to only set this once, the first time?

Reply:
Yes, I think that would be great! Especially when the task spec is not embedded in the taskrun spec, this avoids issues when the task definition is updated during the run.

The only case where is use the change to the spec for something is `cancel`, so we would have to find an alternative for that. Perhaps the `tkn` client could reset the task definition in status to force a reload? @vdemeester what do you think?

Reply:
Oh, actually it would not break `cancel` since that goes in the spec of the taskrun, not of the task - so fetching the task only once would work just fine!

Reply:
yeah